project-koku/koku
# [COST-3315] exclude empty nodes and pvcs when we get them for a cluster

## Jira Ticket

[COST-3315](https://issues.redhat.com/browse/COST-3315)

## Description

This change will prevent us from gathering nodes and pvcs with an empty string name when we get them for a specific cluster.

## Testing

1. On main load in some OCP data, the test customer data flow is perfectly fine for this. `make create-test-customer` followed by `make load-test-customer-data`
2. Go into the database in the `org1234567` schema and insert a record into the `reporting_ocp_nodes` table with a cluster_id that matches one of your sources (replace the cluster_id in this): 
3. `insert into reporting_ocp_nodes (uuid, node, cluster_id) values(gen_random_uuid(), '', '9300d7e2-d474-43e7-916d-9d953a91dc5c');`
4. Go into a python shell with the make command `make shell`
5. Check the `get_nodes_for_cluster` function with that cluster id from before:
```
from masu.database.ocp_report_db_accessor import OCPReportDBAccessor
accessor = OCPReportDBAccessor("org1234567")
accessor.get_nodes_for_cluster('9300d7e2-d474-43e7-916d-9d953a91dc5c')

_outputs something like this_: 
[('aws_compute1', 'i-55555555'), ('aws_compute3', 'i-55555557'), ('aws_compute2', 'i-55555556'), ('aws_master', 'i-55555558'), ('', None)]
```
6. checkout this branch and enter a new python shell the same as step 4
7. repeat step 5 and the output should no longer contain the empty string node:
```
[('aws_compute1', 'i-55555555'), ('aws_compute3', 'i-55555557'), ('aws_compute2', 'i-55555556'), ('aws_master', 'i-55555558')]
```
8. repeat the steps for the `reporting_ocp_pvcs table`
## Notes

...
